src/data/sqlGenerator.ts

```diff
@@ -144,6 +144,11 @@ const generateTraceIdQuery = (options: QueryBuilderOptions): string => {
   if (traceServiceTags !== undefined) {
     selectParts.push(`arrayMap(key -> map('key', key, 'value',${escapeIdentifier(traceServiceTags.name)}[key]), mapKeys(${escapeIdentifier(traceServiceTags.name)})) as serviceTags`);
   }
+  
+  const traceStatusCode = getColumnByHint(options, ColumnHint.TraceStatusCode);
+  if (traceStatusCode !== undefined) {
+    selectParts.push(`if(${escapeIdentifier(traceStatusCode.name)} = 'Error', 2, 0) as statusCode`);
```

Comment:
It would be best if we associated the `Error` value to a versioned otel field. The old version will have `STATUS_CODE_ERROR` while the newer versions have `Error`. We could do this inside of the otel version interface in `otel.ts`

Considering this is a new feature addition, I'll leave this up to you if you want to add it or not

Reply:
Do we still want to add the backwards compatible `OR` here?

Reply:
`OR` was added so I'm resolving this conversation to enable merging.